import{an as t}from"./main-CDRPFaNY.js";import{$ as o}from"./api-DZn48rN0.js";const n=t({id:"record-store",state:()=>({errors:[],successMessage:""}),actions:{async fetchRecord(s=1,r=""){try{return await o("/course-detail",{method:"GET",headers:{"Content-Type":"application/json"},params:{page:s,searchQuery:r}})}catch(e){throw console.error("Error getting application list:",e),this.errors=e.response?e.response.data.errors:["An unexpected error occurred"],e}},async submitApplication(s){try{const r=await o("/course-detail",{method:"POST",body:s,headers:{"Content-Type":"application/json"}});return this.successMessage="Data submitted successfully",r.data}catch(r){throw console.error("Error submitting application:",r),this.errors=r.response?r.response.data.errors:["An unexpected error occurred"],r}},async submitEditApplication(s,r){try{const e=await o(`/course-detail/${s}`,{method:"PUT",body:r,headers:{"Content-Type":"application/json"}});return this.successMessage="Data submitted successfully",e.data}catch(e){throw console.error("Error submitting application:",e),this.errors=e.response?e.response.data.errors:["An unexpected error occurred"],e}},async deleteRecord(s){try{await o(`/course-detail/${s}`,{method:"DELETE"})}catch(r){console.error("Error deleting course details:",r),this.errors=r.response?r.response.data.errors:["An unexpected error occurred"]}}}});export{n as u};
